File: matlab/ch02/ex242.m
% Numerical Integration
f  =  @(x) 1/(1+x^2); a  =  0; b  =  1.5; n  =  10;
z  =  trapzoid(f,a,b,n) % trapezoidal rule
z  =  simps(f,a,b,n) % Simpson 1/3 rule

File: python/ch02/ex242.py
import numpy as np
from trapzoid import trapzoid
from simps import simps

# 1. 함수 정의
f = lambda x: 1 / (1 + x**2)
a, b, n = 0, 1.5, 10

# 결과 실행
z_trap = trapzoid(f, a, b, n)
z_simp = simps(f, a, b, n)

print(f"Trapezoidal rule: {z_trap:.6f}")
print(f"Simpson 1/3 rule: {z_simp:.6f}")